src/ray/core_worker/core_worker.cc

```diff
@@ -1786,6 +1786,68 @@ Status CoreWorker::GetExperimentalMutableObjects(
   return Status::OK();
 }
 
+Status CoreWorker::WaitAndGetExperimentalMutableObjects(
```

Comment:
I feel we should just call it `GetExperimentalMutableObjects`, "get" implies "wait", then this can be combined with the existing `GetExperimentalMutableObjects` method, and we can have an optional `num_objects` arg.

Reply:
`GetExperimentalMutableObjects` will be removed after this PR is merged so that compiled graph doesn't need to share the codepath `get_objects` with other components.